java-frontend/src/main/java/org/sonar/java/bytecode/cfg/BytecodeCFGBuilder.java

```diff
@@ -315,10 +316,10 @@ public void visitTypeInsn(int opcode, String type) {
     public void visitJumpInsn(int opcode, Label label) {
       if(opcode == GOTO || opcode == JSR) {
         currentBlock.terminator = new Instruction(opcode);
-        blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor());
+        currentBlock.successors = Collections.singletonList(blockByLabel.computeIfAbsent(label, l -> currentBlock.createSuccessor()));
```

Comment:
I think it will be cleaner to make `createSuccessor` take a label as an argument.
Here you are setting `successors` in the `createSuccessor`, when the label is missing, and same field is being immediately overwritten, which makes logic bit hard to follow. Also probably all other usage of `computeIfAbsent` could be replaced by such method.